compiler/rustc_span/src/symbol.rs

```diff
@@ -1661,7 +1673,15 @@ impl Ident {
 
     /// Returns `true` if the token is either a special identifier or a keyword.
     pub fn is_reserved(self) -> bool {
-        self.is_special() || self.is_used_keyword() || self.is_unused_keyword()
+        // Note: `span.edition()` is relatively expensive, don't call it unless necessary.
+        self.is_reserved_in_edition(self.span.edition())
```

Comment:
There are three issues here:
- You copy the comment about `span.edition()` being expensive, and then evaluate it eagerly even if it's not necessary.
- Methods taking `Edition` are put on `Ident` and not on `Symbol` despite not using the span.
- The newly added methods duplicate logic from old methods.

I'll try to come up with something better today.

Reply:
Sorry about that. I’m not very good at tinkering with the compiler :/

Reply:
>I'll try to come up with something better today.

Implemented in https://github.com/rust-lang/rust/pull/80226.